### PR TITLE
Add invite_room_state to spec

### PR DESCRIPTION
**Pull request by Kegsay**
Flesh out info.

### CODE REVIEW

File: event-schemas/schema/v1/m.room.member [hunk 1]
```diff
@@ -65,17 +65,22 @@
             "description": "A subset of the state of the room at the time of the invite, if ``membership`` is ``invite``",
             "items": {
                 "type": "object",
-                "title": "StateEvent",
+                "title": "StrippedState",
```
Comment (illicitonion) [inline event-schemas/schema/v1/m.room.member]: Are none of these fields required? From your examples, it looks like they all are, but they're not marked in the schema as such.

File: templating/matrix_templates/units.py [hunk 2]
```diff
@@ -532,6 +532,14 @@ def load_event_schemas(self):
                     Units.prop(json_schema, "properties/content")
                 )
 
+                # this is horrible
```
Comment (illicitonion) [inline templating/matrix_templates/units.py]: Mention why this is necessary (i.e. that we want to extend non-content object properties)

File: event-schemas/schema/v1/m.room.member (hunk 1 above)
Comment (Kegsay) [inline event-schemas/schema/v1/m.room.member]: Done

File: templating/matrix_templates/units.py (hunk 2 above)
Comment (Kegsay) [inline templating/matrix_templates/units.py]: Done

**Comment by illicitonion** [comment]
LGTM

**Pull request merged**